src/app/shared/validators/compare.validator.ts

```diff
@@ -0,0 +1,11 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+export function compareValidator(
+  compareControl: AbstractControl | null
+): ValidatorFn {
+  return (control: AbstractControl): ValidationErrors | null => {
+    const isEqual = control.value === compareControl?.value;
+
+    return !isEqual ? { notEqual: true } : null;
```

Comment:
```suggestion
    return control.value !== compareControl?.value ? { notEqual: true } : null;
```
get rid of unnecessary variable